azure/storage/handler/cloud_queue_client_handler.rb

```diff
@@ -0,0 +1,118 @@
+          reqOptions[:prefix] = listQueuesPayload.prefix unless listQueuesPayload.prefix.nil?
+          reqOptions[:metadata] = true if listQueuesPayload.detailsIncluded == Azure::Storage::AutoGenerated::ThriftQueueListingDetails::Metadata
+          reqOptions[:max_results] = maxResults
+          reqOptions[:marker] = currentToken.nextMarker
+          # Azure::Storage::Infrastructure::LoggingAspect.logger.debug reqOptions
+
+          # ==== Operation ==== #
+          temp = queueClient.list_queues(reqOptions)
+
+          # ==== Construct Reture Value ==== #
+          entries = []
+          temp.each do |queue|
+            entry = Azure::Storage::AutoGenerated::CloudQueueResponse.new
+            entry.name = queue.name
+            entry.metadata = queue.metadata
+            # TODO storageURI
+            entries.push(entry)
+          end
+          result = Azure::Storage::AutoGenerated::ListQueuesResponse.new
+          result.continuationToken.nextMarker = temp.continuation_token
+          result.queueList = entries
+          return result
+        end
+
+        def setProperties(thriftRequestOptions, accountInfo, thriftOperationContext, properties)
+          queueClient = Azure::Storage::Converter::QueueConverter.getQueueService(self, accountInfo)
+          # ==== Construct Parameters ==== #
+          reqOptions = Azure::Storage::Converter::CoreConverter.getCommonRequestOptions(thriftRequestOptions, thriftOperationContext)
+          serviceProperties = Azure::Storage::Converter::CoreConverter.convertThriftStorageServicePropertiesToServiceProperties(properties)
+
+          # ==== Operation ==== #
+          queueClient.set_service_properties(serviceProperties, reqOptions)
+
+          # ==== Construct Reture Value ==== #
+          return
+        end
+        def getProperties(thriftRequestOptions, thriftOperationContext, accountInfo)
+          queueClient = Azure::Storage::Converter::QueueConverter.getQueueService(self, accountInfo)
+          # ==== Construct Parameters ==== #
+          reqOptions = Azure::Storage::Converter::CoreConverter.getCommonRequestOptions(thriftRequestOptions, thriftOperationContext)
+
+          # ==== Operation ==== #
+          queueClient.get_service_properties(reqOptions)
+
+          # ==== Construct Reture Value ==== #
+          result = Azure::Storage::AutoGenerated::ThriftStorageServiceProperties.new
+          result.corsRules
+          result.defaultServiceVersion
+          result.logging
+          result.minuteMetrics
+          result.hourMetrics
+          result.deleteRetentionProperties
+
+          return result
+        end
+        def getServiceStats(thriftRequestOptions, thriftOperationContext, accountInfo)
+          queueClient = Azure::Storage::Converter::QueueConverter.getQueueService(self, accountInfo)
+          # ==== Construct Parameters ==== #
+          # where is the interface of this function
```

Comment:
The function is not yet supported, wait for the newest release of this client library.